monai/transforms/utility/array.py

```diff
@@ -1163,6 +1162,47 @@ def __call__(self, img: NdarrayOrTensor):
         return out
 
 
+class TorchIO(Transform, RandomizableTrait):
+    """
+    This is a wrapper transform for TorchIO transforms based on the specified transform name and args.
+    See https://torchio.readthedocs.io/transforms/transforms.html for more details.
+    """
+
+    backend = [TransformBackends.TORCH]
+
+    def __init__(self, name: str, *args, **kwargs) -> None:
+        """
+        Args:
+            name: The transform name in TorchIO package.
+            args: parameters for the TorchIO transform.
+            kwargs: parameters for the TorchIO transform.
+
+        Note:
+            The `p=` kwarg of TorchIO transforms control set the probability with which the transform is applied.
+            You can specify the probability of applying the transform by passing either `prob` ot `p` in kwargs but'
+            ' not both.
+        """
+        super().__init__()
+        self.name = name
+        transform, _ = optional_import("torchio.transforms", "0.18.0", min_version, name=name)
+
+        if "prob" in kwargs:
+            if "p" in kwargs:
+                raise ValueError("Cannot specify both 'prob' and 'p' in kwargs.")
+            kwargs["p"] = kwargs.pop("prob")
+
+        self.trans = transform(*args, **kwargs)
+
+    def __call__(self, img: NdarrayOrTensor):
+        """
+        Args:
+            img: an instance of torchio.Subject, torchio.Image, numpy.ndarray, torch.Tensor, SimpleITK.Image,
+                 or dict containing 4D tensors as values
+
+        """
+        return self.trans(img)
```

Comment:
If we want to include this transform in the MONAI transform chain, I guess we should ensure the output type which can be compatible with the existing transforms. Perhaps we can add a flag here, what do you think?

Reply:
For the randomised transforms the deterministic interface that we use in MONAI should be used here as well if possible. This includes setting the random state with `randomize` (like [here](https://github.com/Project-MONAI/MONAI/blob/4029c422f1da84cdf5e4ce546dbff33245cbe025/monai/transforms/spatial/array.py#L1475)) and using `set_random_state` to set the state. This may be difficult with how TorchIO transforms work but we can do as much as feasible here.

Reply:
Okay so for all inherently random use RandomizableTrait and for the inherently deterministic ones use the deterministic interface.
Can you provide me a link to the deterministic interface/an example that uses it, please?

Reply:
[RandRotate90](https://github.com/Project-MONAI/MONAI/blob/4029c422f1da84cdf5e4ce546dbff33245cbe025/monai/transforms/spatial/array.py#L1191) may be a good example of how to do things in a simple format. We use `randomize` to compute random state the transform relies on, then apply it in `__call__` as needed. [RandRotate90d](https://github.com/Project-MONAI/MONAI/blob/4029c422f1da84cdf5e4ce546dbff33245cbe025/monai/transforms/spatial/dictionary.py#L672) demonstrates doing this for dictionary based transforms.